example/integration_test/shelley_address_test_cases.dart

```diff
@@ -0,0 +1,521 @@
+import 'package:ledger_cardano/ledger_cardano.dart';
+import 'package:ledger_cardano/src/utils/cardano_networks.dart';
+import 'package:ledger_cardano/src/models/parsed_address_params.dart';
+import 'package:ledger_cardano/src/models/shelley_address_params.dart';
+import 'package:ledger_cardano/src/models/spending_data_source.dart';
+import 'package:ledger_cardano/src/models/staking_data_source.dart';
+
+class ShelleyTestCase {
+  final String testName;
+  final CardanoNetwork network;
+  final ParsedAddressParams addressParams;
+  final String expectedResult;
+
+  ShelleyTestCase({
+    required this.testName,
+    required this.network,
+    required this.addressParams,
+    required this.expectedResult,
+  });
+}
+
+List<ShelleyTestCase> shelleyTestCases = [
+  ShelleyTestCase(
+    testName: 'base address path/path 1',
+    network: CardanoNetwork.custom(customNetworkId: 0x03, customNetworkMagic: 47),
+    addressParams: ParsedAddressParams.shelley(
+      shelleyAddressParams: ShelleyAddressParamsData.basePaymentKeyStakeKey(
+        networkId: 0,
```

Comment:
Like we discussed in the call, we shouldn't sent networkId to address params data